### ISSUE TITLE
Port 1 entered blocking/forwarding/disabled state Issue

### ISSUE THREAD

**Issue by cjpollock** [jayatvars/balena-chromium-kiosk#3]
Hi team,

I'm trying to deploy this code onto my Balena project with a Raspberry Pi 3 B+ and I keep getting a wall of text that states something like:

`eth0: renamed from veth*******`
`port 1 entered blocking state`
`port 1 entered forwarding state`
`port 1 entered disabled state`

and this continues indefinitely. I haven't changed the code at all, and I'm not seeing a ton of documentation regarding this. Any insight?

**Comment by jayatvars** [comment]
Nope. That’s news to me. But the PIs this was tested on were running balenaOS 2.29.2+rev2. Maybe that has something to do with it.


Jay Wilson
Sr. System Administrator
Varsity Spirit

CONFIDENTIALITY NOTICE: This e-mail and any attachments are for the exclusive and confidential use of intended recipient. If you are not the intended recipient, please do not read, distribute or take action in reliance to this message. If you have received this in error, please notify us immediately by return e-mail and promptly delete this message and its attachments from your computer system.

**Comment by cjpollock** [comment]
Interesting. I tried using the balenaOS 2.29.2+rev2 OS and it is still failing with the same on-screen readout, but the Balena logs give me something new:

`Failed to start service 'chromium-kiosk sha256:247b94c3f212b1a00fea2ccd618483f7e34f05295f85f278aaf0d61c2652b706' due to '(HTTP code 400) unexpected - OCI runtime create failed: container_linux.go:296: starting container process caused "exec: \"/usr/bin/entry.sh\": permission denied": unknown '`

I'll do a little digging and report back if I'm able to fix it. Let me know if you think of anything!

**Comment by jacobrosenthal** [comment]
systemd is a good bit of complexity and was deprecated (hence all the complexity readding it here) Ive been using something closer to the x11 example of not using any systemd stuff and only enabling udev
https://github.com/balena-io-playground/x11-window-manager/blob/master/Dockerfile.raspberrypi3#L25

**Comment by cjpollock** [comment]
Gotcha. I added UDEV=1 as an ENV variable and also added your new start.sh file and Dockerfile template from your Systemd fork. It is now loading the default Chrome homepage, but the right half of the window is black, and it doesn't seem to be pulling the URL from URL_LAUNCHER_URL.

Here are some of the errors I'm getting in the log:

`26.07.19 14:17:50 (-0600)  chromium-kiosk  [92:119:0726/201750.369976:ERROR:bus.cc(393)] Failed to connect to the bus: Failed to connect to socket /var/run/dbus/system_bus_socket: No such file or directory
26.07.19 14:17:50 (-0600)  chromium-kiosk  [130:130:0726/201750.941989:ERROR:gl_implementation.cc(280)] Failed to load libGLESv2.so.2: libGLESv2.so.2: cannot open shared object file: No such file or directory
26.07.19 14:17:50 (-0600)  chromium-kiosk  [130:130:0726/201750.961324:ERROR:viz_main_impl.cc(170)] Exiting GPU process due to errors during initialization
26.07.19 14:17:51 (-0600)  chromium-kiosk  [92:319:0726/201751.655107:ERROR:bus.cc(393)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
26.07.19 14:17:52 (-0600)  chromium-kiosk  [1:7:0726/201752.538356:ERROR:command_buffer_proxy_impl.cc(106)] ContextResult::kTransientFailure: Shared memory region is not valid`

**Comment by jacobrosenthal** [comment]
I think thats all log noise iv seen before. One thing, I screwed up the URL parameter and rebase pushed, make sure you have the fix from 2 hours ago firstly

**Comment by jacobrosenthal** [comment]
make sure you gave yourself some extra gpu_mem
gpu_mem=196 in config.txt or change it in the balena fleet configuration dashboard

**Comment by cjpollock** [comment]
Good call; I updated the code with your last edit and now the website is loading correctly. The only issue now is that the screen is black on one half. Looks like a similar log readout, so I am attaching an image:

`26.07.19 16:33:00 (-0600)  chromium-kiosk  [93:120:0726/223300.182433:ERROR:bus.cc(393)] Failed to connect to the bus: Failed to connect to socket /var/run/dbus/system_bus_socket: No such file or directory
26.07.19 16:33:00 (-0600)  chromium-kiosk  [130:130:0726/223300.679975:ERROR:gl_implementation.cc(280)] Failed to load libGLESv2.so.2: libGLESv2.so.2: cannot open shared object file: No such file or directory
26.07.19 16:33:00 (-0600)  chromium-kiosk  [130:130:0726/223300.696745:ERROR:viz_main_impl.cc(170)] Exiting GPU process due to errors during initialization
26.07.19 16:33:01 (-0600)  chromium-kiosk  [93:320:0726/223301.470851:ERROR:bus.cc(393)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")`

![img](https://user-images.githubusercontent.com/52300947/61984802-19551280-afc3-11e9-8f20-48cce8eb7b58.jpg)

**Comment by jacobrosenthal** [comment]
my guess is your hdmi settings in config.txt arent correct? Mine is
something like
## raspberry screen
hdmi_cvt=800 480 60 6 0 0 0
hdmi_drive=1
hdmi_group=2
hdmi_mode=87
hdmi_force_hotplug=1

after that. this repo works for me.. however in another repo I use far more
flags You might try at the very least --start-fullscreen

**Comment by cjpollock** [comment]
The issue persists. I am doing a little digging myself, but it appears other have had this issue as well: https://unix.stackexchange.com/questions/273989/how-can-i-make-chromium-start-full-screen-under-x

**Comment by cjpollock** [comment]
Update: I was able to solve this issue by changing the start.sh file to:

`echo "chromium-browser --kiosk --window-size=1920,1080 --app=$URL_LAUNCHER_URL --disable-dev-shm-usage" >> /home/chromium/xstart.sh`
